### PR TITLE
build JUnit 4 for use by Daikon

### PR DESCRIPTION
**Pull request by markro49**

### CODE REVIEW

**Comment by markro49** [comment]
Copied some of the documentation from typetools/commons-bcel.  I'm sure there are mistakes/omissions in my modifications of build and release instructions.

**Comment by markro49** [comment]
in conjunction with https://github.com/codespecs/daikon/pull/436

File: pom.xml [hunk 1]
```diff
@@ -417,18 +363,32 @@
...
-	      <artifactId>maven-dependency-plugin</artifactId>
-	      <executions>
-		<execution>
-		  <goals>
-		    <goal>properties</goal>
-		  </goals>
-		</execution>
-	      </executions>
-	    </plugin>
+            <plugin>
```
Comment (mernst) [inline pom.xml]: In general I approve of changing tabs to spaces.  However:
 * If you do that, please do so in a separate pull request, to make clear the substantive changes in this pull request.
 * Don't make formatting improvements in a fork of an upstream repository, because it makes the diffs unnecessarily large.

File: pom.xml [hunk 2]
```diff
@@ -56,10 +56,6 @@
         </contributor>
     </contributors>
 
-    <prerequisites>
```
Comment (mernst) [inline pom.xml]: Your change is objectively better, but it introduces additional differences between the upstream JUnit 4 (which has this `<prerequisites>` block) and our version.  I would like to keep diffs to a minimum.

File: pom.xml [hunk 3]
```diff
@@ -609,80 +569,125 @@
                 </plugins>
             </reporting>
         </profile>
+
         <profile>
-            <id>java9</id>
```
Comment (mernst) [inline pom.xml:613]: The `java9` and `java12` profiles disappeared entirely.  Maybe this is the only way, but I wonder whether it is possible to make a change with a smaller diff from upstream.

File: pom.xml [hunk 4]
```diff
@@ -609,80 +569,125 @@
...
+        <!-- This is the base profile for running the Checker Framework.  It will always be run,
+             assuming the Java JDK is > 1.7.  (Which is must be for the Checker to work.) -->
+          <id>checkerframework</id>
+          <activation>
+            <jdk>[1.8,)</jdk>
+          </activation>
+          <build>
+            <plugins>
+              <plugin>
+                <artifactId>maven-compiler-plugin</artifactId>
```
Comment (mernst) [inline pom.xml:712]: Was this moved here fram line 235?  Given the big diffs, any upstream changes to it will cause a merge conflict.
Would it be possible to just add extra arguments there and have smaller diffs?  I'm OK with every compilation always running the Checker Framework.  I'm also OK with not supporting compilation under Java 8 if it is too difficult (especially since there isn't yet a profile just for Java 8 and lower).

If this is not feasible or would be a huge amount of work, we can forego it, since there are unlikely to be many new changes, but the smaller diffs will make the changes easier to understand as well.

File: cfMavenCentral.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+            <name>Michael Ernst</name>
+            <email>mernst@cs.washington.edu</email>
+            <url>http://homes.cs.washington.edu/~mernst/</url>
+            <organization>University of Washington</organization>
+            <organizationUrl>https://www.cs.washington.edu/</organizationUrl>
+        </developer>
+
+        <developer>
+            <id>Makr Roberts</id>
+            <name>Makr Roberts</name>
```
Comment (mernst) [inline cfMavenCentral.xml]: Typo "Makr"

File: README-typetools.txt [hunk 6]
```diff
@@ -0,0 +1,105 @@
+The cuurrent official version of JUnit 4 is junit-4.13.2.jar; released on Feb.13, 2021.
```
Comment (mernst) [inline README-typetools.txt]: Typo "cuurrent"

**Review by mernst — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (markro49) [reply pom.xml]: While the <prerequisites> block does work, it gets a warning which I was trying to eliminate:
[WARNING] The project junit:junit:jar:4.13.2-Daikon uses prerequisites which is only intended for maven-plugin projects but not for non maven-plugin projects.

It's fine to leave it in if you wish.

File: pom.xml (hunk 1 above)
Comment (markro49) [reply pom.xml]: I'm a little confused.  Point (1) is a general comment, but point (2) says don't do it in this case?

Comment (mernst) [reply pom.xml]: Your understanding is correct.  Sorry for the confusion!  I think I already undid the whitespace changes.

File: pom.xml (hunk 3 above)
Comment (markro49) [reply pom.xml:613]: was able to work around this with <?ignore ..... ?>

File: pom.xml (hunk 4 above)
Comment (markro49) [reply pom.xml:712]: was able to clean the diff up with by using <?ignore ... >?

**Review by mernst — APPROVED** [review]
